src/components/analyses/listing/Listing.js

```diff
@@ -125,6 +127,42 @@ function Listing(props) {
         intl,
     ]);
 
+    const updateAnalyses = useCallback(
+        (notifiMessage) => {
+            let push_msg = null;
+            try {
+                push_msg = JSON.parse(notifiMessage);
+            } catch (e) {
+                return;
+            }
+
+            const message = push_msg?.message;
+            if (message) {
+                const category = message.type;
+                let analysisStatus =
+                    category.toLowerCase() ===
+                    constants.NOTIFICATION_CATEGORY.ANALYSIS.toLowerCase()
+                        ? message.payload.status
+                        : "";
+
+                let found = data?.analyses?.find(
+                    (analysis) => analysis.id === message.payload.id
+                );
+
+                if (found && analysisStatus !== found.status) {
+                    found.status = analysisStatus;
+                    found.enddate = message.payload.enddate;
+                    setData({ analyses: [...data?.analyses] });
```

Comment:
I'm not certain if this is actually a problem, but could the `...data?.analyses` part cause issues if data is null or undefined?

Reply:
Hmm.. if `data` or `analyses` is null or undefined, found will be `null` and hence that code wont be executed.